developer/wip/for-contributors/data-plane/custom-data-plane.md

```diff
@@ -0,0 +1,252 @@
+# Writing a custom data plane
+
+<!-- TOC -->
+* [Writing a custom data plane](#writing-a-custom-data-plane)
+  * [1. The Registration Phase](#1-the-registration-phase)
+  * [2. Handling DPS messages ](#2-handling-dps-messages)
+    * [2.1 `START`](#)
+        * [2.1.1 `PUSH`](#211-push)
+        * [2.1.2 `PULL`](#212-pull) 
+    * [2.2 `SUSPEND` and `TERMINATE`](#22-suspend-and-terminate)
+
+<!-- TOC -->
+
+When not running with an embedded data plane embedded, EDC uses the Data Plane Signaling protocol ([DPS](./data-plane-signaling/data-plane-signaling.md)) for the communication between control planes and data planes. In this chapter we will see how to leverage on [DPS](./data-plane-signaling/data-plane-signaling.md) for writing a custom data plane from scratch.
+
+Just as example this chapter contains JS snippet using `express` as web framework.
+
+>  Since it's just for didactic purpose the code is not to be considered complete since proper error and JSON-LD handling are are not implemented.
```

Comment:
```suggestion
>  Since it's only for educational purposes, the code is not intended to be complete, as proper error handling and JSON-LD processing are not implemented
```